CMakeLists.txt

```diff
@@ -11,11 +11,18 @@ find_package(rclcpp REQUIRED)
 find_package(rclcpp_action REQUIRED)
 find_package(Threads REQUIRED)
 
+include_directories(include)
+
 add_library(${PROJECT_NAME} src/realtime_clock.cpp)
 ament_target_dependencies(${PROJECT_NAME} "rclcpp" "rclcpp_action")
 target_link_libraries(${PROJECT_NAME} ${CMAKE_THREAD_LIBS_INIT})
 target_include_directories(${PROJECT_NAME} PUBLIC include)
 
+# A library to detect a realtime kernel and set thread priority, if one is found
+add_library(thread_priority SHARED src/thread_priority.cpp)
```

Comment:
Why did you make this a shared library?

Reply:
i don't have a strong opinion. Execution time shouldn't really matter, so I figured SHARED would be a bit more memory-efficient

Reply:
What part of using a shared library is more memory efficient? Are you trying to reduce the memory used by the linker?

Reply:
**As well as leading to smaller applications**, shared libraries offer the user the ability to use their own, perhaps better version of the libraries rather than relying on one that's part of the application

https://stackoverflow.com/a/2649379/3499467

Again, I don't really care so much. If you have a strong reason to make it a static library, that's fine with me.

Reply:
I think static libraries should be the default unless you are creating a plugin or something like openssl that has the requirements of being updated for security reasons.